app/validators/questionnaire_validator.py

```diff
@@ -134,3 +139,38 @@ def validate_introduction_block(self):
         )
         if not has_introduction_blocks:
             self.add_error(error_messages.PREVIEW_WITHOUT_INTRODUCTION_BLOCK)
+
+    def validate_list_references(self):
+        lists_with_context = self.questionnaire_schema.lists_with_context
+
+        # We need to keep track of section index for: common_definitions.json#/section_enabled
+        for section_index, section in enumerate(self.questionnaire_schema.sections):
+            identifier_references = get_object_containing_key(section, "source")
+            for _, identifier_reference, parent_block in identifier_references:
+                if identifier_reference["source"] == "list":
+                    list_identifier = identifier_reference["identifier"]
+                    if parent_block:
+                        if (
+                            self.questionnaire_schema.block_ids.index(
+                                parent_block["id"]
+                            )
+                            < lists_with_context[list_identifier]["block_index"]
+                        ):
+                            self.add_error(
+                                error_messages.LIST_REFERENCED_BEFORE_ADDED.format(
+                                    list_name=list_identifier
```

Comment:
Might not be possible, but could the parent block ID also be output in this error message, just for some extra info to help narrow down the location in the schema?

Reply:
Not a problem,  I will add it as a key/value.

Reply:
We could possibly keep it as "parent block id" since the block id could be sometimes the nested one, like the add block one.

Reply:
I've added it as "parent block id" for now.